iceoryx_hoofs/include/iceoryx_hoofs/internal/units/duration.inl

```diff
@@ -79,31 +84,35 @@ inline constexpr Duration Duration::fromNanoseconds(const T value) noexcept
     auto clampedValue = positiveValueOrClampToZero(value);
     auto seconds = static_cast<Duration::Seconds_t>(clampedValue / Duration::NANOSECS_PER_SEC);
     auto nanoseconds = static_cast<Duration::Nanoseconds_t>(clampedValue % Duration::NANOSECS_PER_SEC);
-    return Duration{seconds, nanoseconds};
+    return createDuration(seconds, nanoseconds);
 }
 template <typename T>
 inline constexpr Duration Duration::fromMicroseconds(const T value) noexcept
 {
-    auto clampedValue = positiveValueOrClampToZero(value);
-    auto seconds = static_cast<Duration::Seconds_t>(clampedValue / Duration::MICROSECS_PER_SEC);
-    auto nanoseconds = static_cast<Duration::Nanoseconds_t>((clampedValue % Duration::MICROSECS_PER_SEC)
-                                                            * Duration::NANOSECS_PER_MICROSEC);
-    return Duration{seconds, nanoseconds};
+    const auto clampedValue = positiveValueOrClampToZero(value);
```

Comment:
Interesting that it does not complain about lack of `{}` initialization here. Better as is anyway, IMO.